src/checker/zx/ZXChecker.cpp

```diff
@@ -162,26 +179,11 @@ qc::Permutation complete(const qc::Permutation& p, const std::size_t n) {
     mappedTo[j.value()]              = true;
   }
 
-  // Map rest greedily
-
-  for (std::size_t i = 0; i < n; ++i) {
-    if (mappedFrom[i]) {
-      continue;
-    }
-
-    for (std::size_t j = 0; j < n; ++j) {
-      if (!mappedTo[j]) {
-        pComp[static_cast<qc::Qubit>(i)] = static_cast<qc::Qubit>(j);
-        mappedTo[j]                      = true;
-        break;
-      }
-    }
-  }
   return pComp;
 }
 
 qc::Permutation invertPermutations(const qc::QuantumComputation& qc) {
-  return concat(invert(complete(qc.outputPermutation, qc.getNqubits())),
+  return concat(complete(invert(qc.outputPermutation), qc.getNqubits()),
```

Comment:
Could you give me the gist of why swapping the order of operations here helps?

Reply:
Thanks for the explanations! That makes sense and I like the overall workaround for now.